addons/binding/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/HeosChannelHandlerFactory.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.heos.internal;
+
+import static org.openhab.binding.heos.HeosBindingConstants.*;
+
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.openhab.binding.heos.handler.HeosBridgeHandler;
+import org.openhab.binding.heos.internal.api.HeosFacade;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandler;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerBuildGroup;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerControl;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerDynGroupHandling;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerFavoriteSelect;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerGrouping;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerInputs;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerMute;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerPlayURL;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerPlayerSelect;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerPlaylist;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerRawCommand;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerReboot;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerRepeatMode;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerShuffleMode;
+import org.openhab.binding.heos.internal.handler.HeosChannelHandlerVolume;
+
+/**
+ * The {@link HeosChannelHandlerFactory} is responsible for creating the single handler
+ * for the channel of the single things.
+ *
+ * @author Johannes Einig - Initial contribution
+ *
+ */
+
+public class HeosChannelHandlerFactory {
+
+    private HeosBridgeHandler bridge;
+    private HeosFacade api;
+
+    public HeosChannelHandlerFactory(HeosBridgeHandler bridge, HeosFacade api) {
+        this.bridge = bridge;
+        this.api = api;
+    }
+
+    public HeosChannelHandler getChannelHandler(ChannelUID channelUID) {
+        ChannelTypeUID channelTypeUID;
+        Channel channel = bridge.getThing().getChannel(channelUID.getId());
+        if (channel == null) {
+            channelTypeUID = null;
+        } else {
+            channelTypeUID = channel.getChannelTypeUID();
+        }
+        switch (channelUID.getId()) {
+            case CH_ID_CONTROL:
+                return new HeosChannelHandlerControl(bridge, api);
```

Comment:
If I see it correctly, you are creating new instances upon every single `handleCommand` request that is coming in - this looks like a very expensive operation, which is likely to produce a lot of garbage on the heap.